modules/unis.md

```diff
@@ -0,0 +1,153 @@
+MTB-UNI Module
+==============
+
+This specification applies for module types listed below:
+
+| Code   | Module type                                     |
+|--------|-------------------------------------------------|
+| `0x50` | MTB-UNIS with ATmega128 and 6 servo outputs     |
+
+MTB-UNIS module is much like MTB-UNI modules. It contains 16 digital inputs and
+16 digital outputs, and 6 servo outputs. Any of 16 digital output is capable of S-COM protocol
+transmission as well as flickering. Servo is controlled via aditional 12 virtual outputs
+
+## Outputs state
+
+State of each output is encoded in 1 byte:
+
+* `0b 1xxx xxxx`: S-COM code. `x`'s represent specific [S-COM
+  code](https://www.mtb-model.com/elektro/s-com.htm) (0-127).
+* `0b 0000 000x`: digital outputs + servo outputs.
+   - `x=0` = open collector,
+   - `x=1` = output grounded.
+* `0b 0100 xxxx`: flicker output. `xxxx`:
+   - 1: 1 Hz
+   - 2: 2 Hz
+   - 3: 3 Hz
+   - 4: 4 Hz
+   - 5: 5 Hz
+   - 6: 10 Hz
+   - 7: 33 tick/min
+   - 8: 66 tick/min
+
+## Configuration
+
+Configuration consists of 55 bytes. First `24` bytes are same as in MTB-UNI module.
+
+1. 16 bytes of safe outputs state (outputs indexed in order 0 to 15).
+2. 8 bytes of input keep delay.
+   - Byte 0: `0bBBBBAAAA`. `A` = delay of input 0, `B` = delay of input 1.
+   - Byte 1: delay of input 2, delay of input 3.
+   - ...
+   Each input has 16 values of input delay. `0`=0.0s, `1`=0.1s, `2`=0.2s, ...,
+   `15`=1.5s.
+3. 1 byte mask, which servo outputs are active
+   - `0b00654321`, 1 = output active, 0 = output disabled
+4. 24 bytes for servo positions
+   - 2 byte value for servo 1 position 1
+   - 2 byte value for servo 1 position 2
+   - 2 byte value for servo 2 position 1
+   - ...
+5. 6 bytes for servo speed
+   - speed for servo 1
+   - speed for servo 2
+   - ...
+   
+When input goes to logical 0, it must remain in this state for *input keep
+delay* to consider the input as logical 0. Only after the time input changed
+event is reported to master board.
+
+## Module-specific commands
+
+Master → slave:
+
+* *Set Configuration*: n.o. data bytes: 55. Whole configuration is sent.
+* *Get Configuration*: n.o. data bytes: 0.
+* *Get Input*: n.o. data bytes: 0.
+* *Set Output*: n.o. data bytes: variable.
+  - State of all outputs is always sent.
+  - Data byte 0: full state mask for outputs `0b00fedcba`.
+  - Data byte 1: full state mask for outputs `0bFEDCBA98`.
+  - Data byte 2: full state mask for outputs `0b76543210`.
+  - Data byte 3: binary state of outputs `0b00fedcba`.
```

Comment:
Nechceš výstupy pro serva pojmenovat spíš `g-l`? Přijde mi to přehlednější než to, že `f` a `F` jsou různé výstupy.